### PR TITLE
Refactor Orchestrator to handle any combination of extractors, transformers and loaders

### PR DESCRIPTION
**Pull request by JeppeBlixen**
I have a case where i need an etl Orchestrator with multiple extractors and multiple loaders.
This require a change to the Orchestrator.

I think it is redundant to have an Orchestrator class for all combinations of extractors, transformers and loaders.
I have therefore done a refactor of the Orchestrator and OrchestratorBuilder classes to be able to handle any combination of extractors, transformers and loaders.

Changes:
- Refactor SingleOrchestrator, NoTransformOrchestrator, MultipleExtractOrchestrator, MultipleTransformOrchestrator and MultipleLoaderOrchestrator to three classes NoTransformOrchestrator, SingleExtractorOrchestrator and MultipleExtractorOrchestrator that can handle all combinations of extractors, transformers and loaders.
- Refactor OrchestratorBuilder to create correct Orchestrator based on type and number of extractors, transformers and loaders.
- Refactor Orchestrator and OrchestratorBuilder unittests to match changes.
- Add unittests for the OrchestratorBuilderException.
- Add DelegatingMultiInputTransformer to have the possibility of using multiple Transformers with multiple Extractors.
- Refactor Extractor, Transformer and Loader unittests to be independent from Orchestrator.
- Update OETL documentation

### CODE REVIEW

**Review by christianhelle — COMMENTED** [review]
I have a few comments...

File: tests/etl/test_extractor.py [hunk 1]
```diff
@@ -1,59 +1,69 @@
 import unittest
 from unittest.mock import MagicMock
 
-from pyspark.sql import DataFrame
+from pyspark.sql import DataFrame, dataframe
```
[the remark this answers is not in the document]
Comment (JeppeBlixen) [reply tests/etl/test_extractor.py]: Fixed, it was an import typo

**Comment by christianhelle** [comment]
Our docs are very important especially for getting people on-board. Please update them with so they are aligned with the features here

**Comment by christianhelle** [comment]
Nice!

**Pull request merged**